traitsui/qt4/tree_editor.py

```diff
@@ -1323,7 +1323,34 @@ def _menu_new_node(self, class_name, prompt=False):
         node, object, nid = self._data
         self._data = None
         new_node, new_class = self._node_for_class_name(class_name)
-        new_object = new_class()
+        try:
+            new_object = new_class()
+        except TraitError:
+            req_traits_dict = new_class.class_traits(required=True)
+            kwargs = {
+                trait_name: ctrait.default
+                for trait_name, ctrait in req_traits_dict.items()
+            }
+            new_object = new_class(**kwargs)
+
+            prompt = True
```

Comment:
I set prompt to true here as a way to try to encourage the user to adjust/make any corrections that are needed since we use the defaults for required traits (may be overkill and I can undo this change)